From 46cc79c43a95108967ffa041145b8117b5e49589 Mon Sep 17 00:00:00 2001
From: Edward Hervey <edward@centricular.com>
Date: Fri, 21 Sep 2018 16:01:53 +0200
Subject: [PATCH 60/72] avviddec: Remove unneeded check

frame is always valid in this function (but wasn't before the
refactoring from a few months ago).

CID #1439540
---
 ext/libav/gstavviddec.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/ext/libav/gstavviddec.c b/ext/libav/gstavviddec.c
index de23954..8c9a794 100644
--- a/ext/libav/gstavviddec.c
+++ b/ext/libav/gstavviddec.c
@@ -1800,14 +1800,12 @@ gst_ffmpegviddec_handle_frame (GstVideoDecoder * decoder,
   if (!packet.size)
     goto done;
 
-  if (frame) {
-    /* save reference to the timing info */
-    ffmpegdec->context->reordered_opaque = (gint64) frame->system_frame_number;
-    ffmpegdec->picture->reordered_opaque = (gint64) frame->system_frame_number;
+  /* save reference to the timing info */
+  ffmpegdec->context->reordered_opaque = (gint64) frame->system_frame_number;
+  ffmpegdec->picture->reordered_opaque = (gint64) frame->system_frame_number;
 
-    GST_DEBUG_OBJECT (ffmpegdec, "stored opaque values idx %d",
-        frame->system_frame_number);
-  }
+  GST_DEBUG_OBJECT (ffmpegdec, "stored opaque values idx %d",
+      frame->system_frame_number);
 
   /* This might call into get_buffer() from another thread,
    * which would cause a deadlock. Release the lock here
-- 
2.17.0

